app/main/lib/shared_models/audio_model.py

```diff
@@ -212,14 +207,28 @@ def get_context_for_search(self, task):
         return context
 
     def search(self, task):
-        audio, temporary = self.get_audio(task)
-        context = self.get_context_for_search(task)
+        if "body" in task:
+            body = task.get("body", {})
+            threshold = task.get("raw", {}).get('threshold', 0.0)
+            limit = body.get("raw", {}).get("limit")
+        else:
+            body = task
+            threshold = body.get('threshold', 0.0)
+            limit = body.get("limit")
+        audio, temporary = self.get_audio(body)
+        context = self.get_context_for_search(body)
+        if audio.chromaprint_fingerprint is None:
+            callback_url =  Presto.add_item_callback_url(app.config['ALEGRE_HOST'], "audio")
+            if task.get("doc_id") is None:
+                task["doc_id"] = str(uuid.uuid4())
+            response = json.loads(Presto.send_request(app.config['PRESTO_HOST'], "audio__Model", callback_url, task).text)
+            result = Presto.blocked_response(response, "audio")
```

Comment:
Do we have to block here? I was thinking/hoping we can return a generic response to the caller at this point like: ```{"result": "pending", "doc_id": f"task['doc_id']"}```
When Presto callsback, the callback handler will recall search and `audio.chromaprint_fingerprint` will no longer be `None`

Reply:
I think lets just block - this is easier to deal with for now IMO

Reply:
I'm fine to block here, but I believe that means the HTTP request between Check API and Alegre will be held open for the duration of this block. We may get an API Gateway Timeout if the roundtrip exceeds one minute. Having the block here means we're not using the Check API callback URL (Check API experiences a blocking call). I'm OK with that, but I felt we wanted to avoid it wherever possible. It shouldn't be too much to avoid the blocking call here, but we can leave that as future work.